### PR TITLE
Document BuildInstance, BuildCPU, BuildMemory

### PR DESCRIPTION
**Pull request by soulshake**
## Release Playbook
- [ ] Rebase against master
- [ ] Code review
- [ ] Merge into master
- [ ] Verify changes at http://site-staging.convox.com
- [ ] Promote release on `site-production` in the `convox/production` Rack

### CODE REVIEW

**Comment by soulshake** [comment]
[Document build instances, BuildMemory](https://trello.com/c/9qGv0dlC/1962-document-build-instances-buildmemory)

File: _deployment/builds.md [hunk 1]
```diff
@@ -74,3 +74,15 @@ $ convox builds export <build ID> -a <app1> --rack <rack1> | convox builds impor
 ## Build arguments
 
 Convox respects the `ARG` Dockerfile directive. For more information, see [Dockerfile: ARG](/docs/dockerfile/#arg).
+
+## Dedicated build instance
```
Comment (nzoschke) [inline _deployment/builds.md]: I made a similar PR with some of this information here: https://github.com/convox/site/pull/600

It's a matter of style, but I'd like to keep the main docs under Deployment (Installing Rack, Creating an App, Deploying, Builds and Release) as simple as possible. Just cover the basic commands for the the default and happy path.

Then link to other docs with all the advanced options.

The "Automatic Builds" and "Build Args" section get this just right, so a "Dedicated Build Instance" section should follow suit.

So I suggest:

- [ ] Move "Dedicated Build Instance" to the bottom of the doc.
- [ ] Add a high level reason why you might need a dedicated build instance. I.e. "If you need better build performance, you can add a dedicated build instance that will give more CPU, memory and caching to builds. See the Advanced Build Options Doc"

File: _deployment/builds.md [hunk 2]
```diff
@@ -74,3 +74,15 @@ $ convox builds export <build ID> -a <app1> --rack <rack1> | convox builds impor
...
+
+You can assign a dedicated instance to a Rack for building images.
+
+You can set this:
+
+- on a new Rack by passing the `--build-instance` flag to `convox install`,
+- on new Racks by setting the [`RACK_BUILD_INSTANCE`](/docs/cli-environment-variables/#rackbuildinstance) environment variable,
+- on an existing Rack by setting the [`BuildInstance`](/docs/rack-parameters/#buildinstance) Rack parameter
+
+You can customize the amount of memory and CPU allocated to build instances by setting the [`BuildCpu`](/docs/rack-parameters/#buildcpu) and [`BuildMemory`](/docs/rack-parameters/#buildmemory) Rack parameters.
```
Comment (nzoschke) [inline _deployment/builds.md]: Does BuildCpu and BuildMemory apply to the build instance? I think that might be more applicable to the normal case and tune how much CPU and Memory build tasks get inside a standard cluster.

**Comment by nzoschke** [comment]
@soulshake I left some feedback. Can you also help review and get https://github.com/convox/site/pull/600 out along with this?

File: _reference/rack-parameters.md [hunk 3]
```diff
@@ -85,12 +88,32 @@ Override the default build image.
 
 | Default value | "" |
 
+## BuildInstance
+
+Creates and assigns a [dedicated build instance](/docs/builds/#dedicated-build-instance) to a Rack.
+
+The build instance be fine-tuned with the [`BuildCpu`](#buildcpu) and [`BuildMemory`](#buildmemory) Rack parameters.
```
Comment (nzoschke) [inline _reference/rack-parameters.md]: I haven't tested any of this but I don't understand why you'd want to tune this at all on a build instance. With a dedicated instance I'd expect you'd be fine with letting a build consume all the CPU and Memory it wants

Comment (soulshake) [inline _reference/rack-parameters.md]: Me neither, but it's an available Rack parameter, so shouldn't it be documented?

File: _deployment/builds.md (hunk 2 above)
Comment (soulshake) [inline _deployment/builds.md]: Ok, I'll take it out.

File: _reference/rack-parameters.md (hunk 3 above)
Comment (soulshake) [inline _reference/rack-parameters.md]: Disregard previous comment; I think I see what you mean now.

File: _management/advanced-build-options.md [hunk 4]
```diff
@@ -1,6 +1,7 @@
 ---
 title: "Advanced Build Options"
 order: 500
+published: false
```
Comment (nzoschke) [inline _management/advanced-build-options.md]: need to flip this back to published

**Comment by nzoschke** [comment]
@soulshake I removed the `published: false` flag. Can you merge and deploy?

**Pull request merged**